openmetadata-ui/src/main/resources/ui/src/utils/EntityLineageUtils.tsx

```diff
@@ -515,3 +529,292 @@ export const getNodeRemoveButton = (onClick: () => void) => {
+    source,
+    entity.id
+  );
+
+  const targetDownStreamNode = getSourceTargetNode(
+    updatedLineageData.downstreamEdges || [],
+    target,
+    entity.id
+  );
+
+  const targetUpStreamNode = getSourceTargetNode(
+    updatedLineageData.upstreamEdges || [],
+    target,
+    entity.id
+  );
+
+  const isUpstream =
+    (!isNil(sourceUpStreamNode) && !isNil(targetDownStreamNode)) ||
+    !isNil(sourceUpStreamNode) ||
+    !isNil(targetUpStreamNode) ||
+    target?.includes(entity.id);
+
+  const isDownstream =
+    (!isNil(sourceDownstreamNode) && !isNil(targetUpStreamNode)) ||
+    !isNil(sourceDownstreamNode) ||
+    !isNil(targetDownStreamNode) ||
+    source?.includes(entity.id);
+
+  if (isUpstream) {
+    return EdgeTypeEnum.UP_STREAM;
+  } else if (isDownstream) {
+    return EdgeTypeEnum.DOWN_STREAM;
+  }
+
+  return EdgeTypeEnum.NO_STREAM;
+};
+
+/**
+ * Get updated Edge with lineageDetails
+ * @param edges Array of Edge
+ * @param params new connected edge
+ * @param lineageDetails updated lineage details
+ * @returns updated edge array
+ */
+export const getUpdatedEdge = (
+  edges: EntityLineageEdge[],
+  params: Edge | Connection,
+  lineageDetails: LineageDetails | undefined
+) => {
+  const updatedEdge: EntityLineageEdge[] = [];
+  const { target, source } = params;
+  edges.forEach((edge) => {
+    if (edge.fromEntity === source && edge.toEntity === target) {
+      updatedEdge.push({
+        ...edge,
+        lineageDetails: lineageDetails,
+      });
+    } else {
+      updatedEdge.push(edge);
+    }
+  });
+
+  return updatedEdge;
+};
+
+// create new edge
+export const createNewEdge = (
+  params: Edge | Connection,
+  isEditMode: boolean,
+  sourceNodeType: string,
+  targetNodeType: string,
+  isColumnLineage: boolean,
+  onEdgeClick: (
+    evt: React.MouseEvent<HTMLButtonElement>,
+    data: CustomEdgeData
+  ) => void
+) => {
+  const { target, source, sourceHandle, targetHandle } = params;
+  let data: Edge = {
+    id: `edge-${source}-${target}`,
+    source: `${source}`,
+    target: `${target}`,
+    type: isEditMode ? 'buttonedge' : 'custom',
+    style: { strokeWidth: '2px' },
+    markerEnd: {
+      type: MarkerType.ArrowClosed,
+    },
+    data: {
+      id: `edge-${source}-${target}`,
+      source: source,
+      target: target,
+      sourceType: sourceNodeType,
+      targetType: targetNodeType,
+      isColumnLineage: isColumnLineage,
+      onEdgeClick,
+    },
+  };
+
+  if (isColumnLineage) {
+    data = {
+      ...data,
+      id: `column-${sourceHandle}-${targetHandle}-edge-${source}-${target}`,
+      sourceHandle: sourceHandle,
+      targetHandle: targetHandle,
+      style: undefined,
+      data: {
+        ...data.data,
+        id: `column-${sourceHandle}-${targetHandle}-edge-${source}-${target}`,
+        sourceHandle: sourceHandle,
+        targetHandle: targetHandle,
+      },
+    };
+  }
+
+  return data;
+};
+
+export const LoadingStatus = (
```

Comment:
The function name should be in `camelCase`.
```suggestion
export const getLoadingStatus = (
```